scripts/git/google_to_main.sh

```diff
@@ -31,28 +31,12 @@ set -o pipefail
 
 GREEN_COMMIT="${1:-google}"
 
+export UPSTREAM_REMOTE="${UPSTREAM_REMOTE:-upstream}"
 PR_BRANCH="${PR_BRANCH:-google-to-main}"
-UPSTREAM_REMOTE="${UPSTREAM_REMOTE:-upstream}"
 FORK_REMOTE="${FORK_REMOTE:-origin}"
 
-if [[ -n "$(git status --porcelain)" ]]; then
-  echo "Working directory not clean. Aborting"
-  git status
-  exit 1
-fi
-if ! git symbolic-ref -q HEAD; then
-  echo "In a detached HEAD state. Aborting"
-  git status
-  exit 1
-fi
-git checkout google
-git pull "${UPSTREAM_REMOTE?}" google --ff-only
-git submodule update --init
-if [[ -n "$(git status --porcelain)" ]]; then
-  echo "Working directory not clean after sync. Aborting"
-  git status
-  exit 1
-fi
+./scripts/git/git_update.sh main
```

Comment:
Looks like we could get by without actually updating the local branch and just running
```
git fetch "${UPSTREAM_REMOTE}" google
...
git log "${UPSTREAM_REMOTE}"/google..
```

Actually updating the branches is generally useful, but when just running one script on its own it would be nice if it used a minimal number of operations - `git submodule update` can be slow.

WDYT?

Moving the common logic into `git_update.sh` is still a general improvement for these scripts though, can keep that regardless.

Reply:
Fixed. Plus some other things I found along the way.